clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordAccumulator.java

```diff
@@ -272,55 +281,87 @@ private boolean isMuted(TopicPartition tp, long now) {
         return result;
     }
 
+    private void maybeUpdateNextBatchExpiryTime(ProducerBatch batch) {
+        if (batch.createdMs + deliveryTimeoutMs  > 0) {
+            // the non-negative check is to guard us against potential overflow due to setting
+            // a large value for deliveryTimeoutMs
+            nextBatchExpiryTimeMs = Math.min(nextBatchExpiryTimeMs, batch.createdMs + deliveryTimeoutMs);
+        } else {
+            log.warn("Skipping next batch expiry time update due to addition overflow: "
```

Comment:
`batch.createdMs + deliveryTimeoutMs ` should be positive.  we log an warning here due to addition overflow. we have defined `deliverTimeoutMs` as `int` when we initialize the producer. Given that, i am not expecting that that log line will be trigged often. 

       private static int configureDeliveryTimeout(ProducerConfig config, Logger log) {
            int deliveryTimeoutMs = config.getInt(ProducerConfig.DELIVERY_TIMEOUT_MS_CONFIG);
            int lingerMs = config.getInt(ProducerConfig.LINGER_MS_CONFIG);
            int requestTimeoutMs = config.getInt(ProducerConfig.REQUEST_TIMEOUT_MS_CONFIG);

Currently I keep RecordAccumulator constructor signature unchanged. If we change the type of 'deliveryTimeoutMs', it is better to also change the type of `lingerMs` and `retryBackoffMs` to be consistent. As changing the types of those paramters will require changes in many test cases to pass `int` instead of `long`,  I'd rather to put the RecordAccumulator constructor signature change in other PR. 


        public RecordAccumulator(LogContext logContext,
                                 int batchSize,
                                 CompressionType compression,
                                 long lingerMs,
                                 long retryBackoffMs,
                                 long deliveryTimeoutMs,
                                  ...
.

Reply:
Fair enough. I missed the long coercion. I think the log message has dubious value, but I agree we will shouldn't ever actually see it.